### PR TITLE
Add test workflows via GitHub actions

### PR DESCRIPTION
**Pull request by auria**

### CODE REVIEW

File: .github/workflows/tests.yml
Comment (r-f-g) [inline .github/workflows/tests.yml:44]: I think you should use `actions-operator` to be sure that juju and tox are installed. At the same time it provides the possibility to use GitHub secrets for the credentials and not directly.

**Review by r-f-g — COMMENTED** [review]
only one comment

File: .github/workflows/tests.yml
Comment (auria) [reply .github/workflows/tests.yml:44]: `actions-operator` is not in the marketplace, and I'm trying to limit which actions can be set up by limiting to those created by GitHub or verified within the marketplace.

OTOH, the secrets I created in the main repository are not available to forked repositories that create a PR against the main repo. This is the reason why I switched to sourcing a file with credentials that are not directly usable for anybody outside of this test env. This together with the limitation that no tests will be run for new contributors until a maintainer approves it should reduce the risks.

**Review by r-f-g — APPROVED** [review]
LGTM

**Pull request merged**
